File: provider/code-gen-service-8105/src/main/java/com/llc/springcloud/codegen/controller/IndexController.java
package com.llc.springcloud.codegen.controller;

import com.alibaba.fastjson.JSONObject;
import com.llc.springcloud.codegen.tpl.TemplateUtil;
import com.llc.springcloud.codegen.tpl.tpldto.GenConfigDto;
import com.llc.springcloud.codegen.xml.MybatisGenConfigParser;
import com.llc.springcloud.web.response.JsonResponse;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.io.OutputStreamWriter;

@RestController
public class IndexController {
	
	@RequestMapping(value = "index", method = RequestMethod.GET)
	public JsonResponse<JSONObject> index() {
		return JsonResponse.ok(MybatisGenConfigParser.getInstance().parse(null));
	}
	
	@RequestMapping(value = "freemarker", method = RequestMethod.GET)
	public void getFreemarker(HttpServletResponse response) {
		
		GenConfigDto genConfigDto = new GenConfigDto();
		
		try {
			response.setContentType("text/plain");
			TemplateUtil.getInstance()
					.out(new OutputStreamWriter(response.getOutputStream()))
					.render(genConfigDto, "gen-config.ftl");
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	
}

File: provider/kotlin-service-demo-8104/src/main/java/com/llc/springcloud/kotlinservice/controller/IndexController.kt
package com.llc.springcloud.kotlinservice.controller

import com.llc.springcloud.kotlinservice.dto.IndexParamsDto
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController
import javax.servlet.http.HttpServletRequest

@RestController
class IndexController {

	@RequestMapping("index", method = [RequestMethod.GET])
	fun index(request: HttpServletRequest, dto: IndexParamsDto): String {
		return "parent params : ${dto.name}; params: ${dto.userName} ${dto.password}"
	}
}